strawberry/field.py

```diff
@@ -210,10 +210,20 @@ def _get_field(
         if key not in ["info", "return"]
     }
 
-    arguments = {
-        to_camel_case(name): get_graphql_type_for_annotation(annotation, name)
-        for name, annotation in arguments_annotations.items()
-    }
+    if wrap.__defaults__:
+        default_argument_start_index = len(annotations) - len(wrap.__defaults__) - 1
```

Comment:
what do you think of extracting this into an utility function? Feels a bit hard to understand right now, we could have something like: `def is_argument_optional(func: Callable, argument_name: str) -> bool: ...`, it should also prevent the duplication of code :)

Have you considered using `inspect.signature`? I did a quick test and seems also another way of checking if an argument has a default value:

![image](https://user-images.githubusercontent.com/667029/59129313-683bdf80-8964-11e9-9da1-8242d79789a3.png)

Reply:
Oh cool, I didn't know about this method! Ill re-write the logic using `inspect.signature` - it looks like it would be a lot cleaner.